sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp

```diff
@@ -70,24 +70,24 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
 
     /**
      * @brief Create a DataLakePathClient from current DataLakeFileSystemClient
-     * @param path Path of the resource within the file system.
+     * @param path Name of path within the file system.
      * @return DataLakePathClient
      */
     DataLakePathClient GetPathClient(const std::string& path) const;
 
     /**
      * @brief Create a DataLakeFileClient from current DataLakeFileSystemClient
-     * @param path Path of the file within the file system.
+     * @param fileName Name of the file within the file system.
      * @return DataLakeFileClient
      */
-    DataLakeFileClient GetFileClient(const std::string& path) const;
+    DataLakeFileClient GetFileClient(const std::string& fileName) const;
 
     /**
      * @brief Create a DataLakeDirectoryClient from current DataLakeFileSystemClient
-     * @param path Path of the directory within the file system.
+     * @param directoryName Name of the directory within the file system.
      * @return DataLakeDirectoryClient
      */
-    DataLakeDirectoryClient GetDirectoryClient(const std::string& path) const;
+    DataLakeDirectoryClient GetDirectoryClient(const std::string& directoryName) const;
```

Comment:
This is different how it's implemented in .Net. They only have `GetRootDirectoryClient`.

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs#L469
I don't believe what you said is true.
The only thing not align is that .NET is not having GetPathClient. I don't think this inconsistency is being raised during the review so I'm sticking with this design now.

Reply:
okay, I missed that. So we need `GetRootDirectoryClient`.

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs#L496
.NET has this as internal. I don't think we should add that if we don't need that internally.

Reply:
Then it's okay

Reply:
But why don't they have `GetPathClient` anywhere?

Reply:
Maybe they don't think it is important. But I tend to believe it is valuable for the customer to be able to manage the resource without knowing its type. If you take a look at the APIs supported by the path client, you would find that it is overkill to require the customer to identify the type of the resource before the API is called. e.g. GetProperties/SetMetadata.

Reply:
Also, if `PathClient` supports initializing without the type of resource it points to, it is natural for its parent resource to support creating it.